spinn_machine/machine.py

```diff
@@ -287,6 +287,69 @@ def get_global_xy(self, local_x, local_y, ethernet_x, ethernet_y):
         :return: global (x,y) coordinates of the chip
         """
 
+    @abstractmethod
+    def shortest_path_length(self, source, destination):
+        """
+        Get the mathematical length of a shortest path from source to
+        destination
+
+        This method does not check if the chips and links it assumes to take
+        actually exist.
+        For example long paths along a none wrapping edge may well travel
+        through the missing area.
```

Comment:
As a point of interest, if this returned the *actual* shortest path i.e. taking into account the missing chips, it the algorithm that makes use of this a) wouldn't need to fix up the path afterwards and b) might be better since it would get the actual shortest path in the first place, rather than the shortest path if everything works, but as it doesn't a path that is longer than it should be.

Reply:
The shortest path length is a COST function to pick which path to try so needs to be fast and does not need to be accurate.

shortest_path returning an actual path is an interesting one.  One one side yes if could avoiding need to fix. On the other side it pushes the algorithm on how to correct possible too high.

Would suggest leaving this kind of thing until 6.0.0 if at all,

Reply:
OK